src/odemis/acq/stream/_static.py

```diff
@@ -826,13 +826,20 @@ def _updateCalibratedData(self, data=None, bckg=None, coef=None):
         Try to update the data with a new calibration. The two parameters are
         the same as apply_spectrum_corrections(). The input data comes from
         .raw and the calibrated data is saved in .calibrated
+        :param data: (DataArray or None) The raw data. If None is given, .raw[0] is used.
         :param bckg: (DataArray or None) The background image.
         :param coef: (DataArray or None) The spectrum efficiency correction data.
         :raise ValueError: If the data and calibration data are not valid or
           compatible. In that case the current calibrated data is unchanged.
         """
         if data is None:
             data = self.raw[0]  # only one image in .raw for spectrum, temporal spectrum and chronograph
+            prev_bckg, prev_coef = self._calib_parameters
+            # Compare numpy arrays with just "is" to avoid comparing the internal,
+            # which is slow, and because we can assume that arrays are used as read-only.
+            if bckg is prev_bckg and coef is prev_coef:
+                logging.debug("Skipping re-calibration with the same parameters")
+                return
```

Comment:
I thought comparing numpy arrays with "is" does not work. I might just be misunderstanding the code and/or the comment
![image](https://github.com/delmic/odemis/assets/46314774/a8dba852-645d-49c7-90cd-b21ac5d13807)

Reply:
That's what I tried to explained in the comment. We don't compare the array's content. We only check if it's the same "block of memory" that was passed before. For this code, it's very unlikely that we receive a different array, but which turns out to contain the same data. In that very unlikely case, then it'll be a little less quick. No big deal.
Any suggestion for a better comment?

Reply:
From the comment I did not understand that you were trying to check if it had not updated at all. Now that I understand it, the comment makes sense, which makes it harder to suggest an improvement :grimacing:. How about:
```
# Because we can assume that arrays are used as read-only, compare numpy arrays with just "is" to 
# check that they are exactly the same object. Using "is" is faster than comparing the internal array.
```

Reply:
you resolved the comment, but I don't see any changes, is that correct?

Reply:
Sorry... forgot to push :cry: 
Thanks for reminding me!